chain/chain/src/resharding/resharding_actor.rs

```diff
@@ -0,0 +1,23 @@
+use super::types::FlatStorageSplitShardRequest;
+use near_async::messaging::{self, Handler};
+
+/// Dedicated actor for resharding V3.
+pub struct ReshardingActor {}
+
+impl messaging::Actor for ReshardingActor {}
+
+impl Handler<FlatStorageSplitShardRequest> for ReshardingActor {
+    fn handle(&mut self, msg: FlatStorageSplitShardRequest) {
+        self.handle_flat_storage_split_shard_request(msg);
+    }
+}
+
+impl ReshardingActor {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn handle_flat_storage_split_shard_request(&mut self, msg: FlatStorageSplitShardRequest) {
+        msg.resharder.split_shard_task();
```

Comment:
I'm honestly not too happy that we are calling a function from resharder here.

For the purposes of reducing mental burden of identifying what happens where, ideally all the logic to deal with the heavy task of splitting should have been a part of the ReshardingActor class. This is too much of indirection to deal with.

Is it possible for us to push the logic of splitting the flat storage here?

Reply:
I thought the general sentiment was to keep the actor as simple as possible, thus all the work and unit testing is done in the resharder. Any other thoughts?

Reply:
Would love input from other folks, @wacban @Longarithm!

Reply:
I don't have strong feelings either way. My only recommendation would be to keep the actor api part (typically the handle method) as slim as possible. This way we can properly unit test the actual implementation without needed to spin up the actor or worry about async. So something like this:

```
fn handle(request) {
  the_actual_implementation(request); 
} 

mod tests {
  // here the the_actual_implementation can be unit tested without needing to bootstrap an entire actor
```